src/masternode-sync.cpp

```diff
@@ -353,8 +357,13 @@ void CMasternodeSync::Process()
 
                 if (RequestedMasternodeAttempt >= MASTERNODE_SYNC_THRESHOLD * 3) return;
 
-                CBlockIndex* pindexPrev = chainActive.Tip();
-                if (pindexPrev == NULL) return;
+                {
+                    TRY_LOCK(cs_main, lockMain);
+                    if (!lockMain) return;
+
+                    CBlockIndex *pindex = chainActive.Tip();
+                    if (pindex == NULL) return;
+                }
```

Comment:
why this? it seems we don't even need this pointer here.

Reply:
It's just a chain validation in the mn winners list sync to not store it if the node it's in an invalid state. If pindex is null, it's because the chain is not sync.

Still, we should try to contain ourselves from any deep change here in this pre-4.0 period. Let's see if we can do it or well, prepare your knifes because this will end up being a code butchery.